### DISCUSSION TITLE
Query Regarding CSE611

### DISCUSSION THREAD

**Discussion by mohith01** [snigi-gupta/AttendingUniversityAtBuffalo#66] [General]
Hello Snigdha, Thank you for the amazing guide. In this guide, you have mentioned that for a student joining straight from UG, it's better to take course CSE611. 

I would like to know whether it is recommended to take CSE611 immediately after joining or will it be available after spring also for people who joined straight after undergrad?

Thank you

**Comment by snigi-gupta** [comment]
@mohith01 
You should definitely take a 611 project since you will be working on a hands-on project. Try to form a great team (because that matters), and when the companies pitch the projects, make up your mind quickly and grab the project you want to work on. The essential thing is you work with a good client and a good team. The effort for 611 is quite much, so people usually opt for this in their last semester, or second last semester, and I would advise the same. Alan Hunt is a great professor, and if you deliver the deliverables on time, you get a good grade too. This course will take you through the entire lifecycle of real work problems and how to solve them. For a new grad, I highly recommend this course since it will be a great thing to put in your resume. 
If you have follow-up questions, do post them here, and I can answer them!
